File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    cloth_dic = {i[1]:1 for i in clothes}
    
    for cloth in clothes:
        cloth_name, cloth_type = cloth
        
        cloth_dic[cloth_type] += 1
    
    answer = 1
    for i in cloth_dic.values():
        answer *= i
        
    return answer - 1
        

File: 프로그래머스/2/42578. 의상/의상.js
function solution(clothes) {
    var selection = 1;
    var cloth = new Map();
    
    clothes.forEach((val) => {
        if(cloth.get(val[1]) === undefined) cloth.set(val[1], 2)
        else cloth.set(val[1], cloth.get(val[1]) + 1)
        
    })
    
    for(let [key, val] of cloth){ 
        selection *= val
    }
    
    return selection - 1;
}